src/System.Net.Security/src/System/Net/SecureProtocols/_SslStream.cs

```diff
@@ -0,0 +1,709 @@
+        private byte[] _InternalBuffer;
+        private bool _InternalBufferFromPinnableCache;
+
+        private byte[] _PinnableOutputBuffer;                        // Used for writes when we can do it. 
+        private byte[] _PinnableOutputBufferInUse;                   // Remembers what UNENCRYPTED buffer is using _PinnableOutputBuffer.
+
+        private int _InternalOffset;
+        private int _InternalBufferCount;
+
+        private FixedSizeReader _Reader;
+
+        internal _SslStream(SslState sslState)
+        {
+            if (PinnableBufferCacheEventSource.Log.IsEnabled())
+            {
+                PinnableBufferCacheEventSource.Log.DebugMessage1("CTOR: In System.Net._SslStream.SslStream", this.GetHashCode());
+            }
+            _SslState = sslState;
+            _Reader = new FixedSizeReader(_SslState.InnerStream);
+        }
+
+        // If we have a read buffer from the pinnable cache, return it.
+        private void FreeReadBuffer()
+        {
+            if (_InternalBufferFromPinnableCache)
+            {
+                s_PinnableReadBufferCache.FreeBuffer(_InternalBuffer);
+                _InternalBufferFromPinnableCache = false;
+            }
+            _InternalBuffer = null;
+        }
+
+        ~_SslStream()
+        {
+            if (_InternalBufferFromPinnableCache)
+            {
+                if (PinnableBufferCacheEventSource.Log.IsEnabled())
+                {
+                    PinnableBufferCacheEventSource.Log.DebugMessage2("DTOR: In System.Net._SslStream.~SslStream Freeing Read Buffer", this.GetHashCode(), PinnableBufferCacheEventSource.AddressOfByteArray(_InternalBuffer));
+                }
+                FreeReadBuffer();
+            }
+            if (_PinnableOutputBuffer != null)
+            {
+                if (PinnableBufferCacheEventSource.Log.IsEnabled())
+                {
+                    PinnableBufferCacheEventSource.Log.DebugMessage2("DTOR: In System.Net._SslStream.~SslStream Freeing Write Buffer", this.GetHashCode(), PinnableBufferCacheEventSource.AddressOfByteArray(_PinnableOutputBuffer));
+                }
+                s_PinnableWriteBufferCache.FreeBuffer(_PinnableOutputBuffer);
+            }
+        }
+
+        internal int Read(byte[] buffer, int offset, int count)
+        {
+            return ProcessRead(buffer, offset, count, null);
+        }
+
+        internal void Write(byte[] buffer, int offset, int count)
+        {
+            ProcessWrite(buffer, offset, count, null);
+        }
+
+        internal bool DataAvailable
+        {
+            get { return InternalBufferCount != 0; }
+        }
+
+        private byte[] InternalBuffer
+        {
+            get
+            {
+                return _InternalBuffer;
+            }
+        }
+
+        private int InternalOffset
+        {
+            get
+            {
+                return _InternalOffset;
+            }
```

Comment:
I'm not surprised anymore: too much manual work has been required to format code. 
Since this is clear enough I won't do the changes here.